src/AspNet.Security.OAuth.Kook/KookAuthenticationHandler.cs

```diff
@@ -0,0 +1,89 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace AspNet.Security.OAuth.Kook;
+
+public partial class KookAuthenticationHandler : OAuthHandler<KookAuthenticationOptions>
+{
+    public KookAuthenticationHandler(
+        [NotNull] IOptionsMonitor<KookAuthenticationOptions> options,
+        [NotNull] ILoggerFactory logger,
+        [NotNull] UrlEncoder encoder,
+        [NotNull] ISystemClock clock)
+        : base(options, logger, encoder, clock)
+    {
+    }
+
+    protected override async Task<AuthenticationTicket> CreateTicketAsync(
+        [NotNull] ClaimsIdentity identity,
+        [NotNull] AuthenticationProperties properties,
+        [NotNull] OAuthTokenResponse tokens)
+    {
+        using var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+        request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
+
+        using var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+        if (!response.IsSuccessStatusCode)
+        {
+            await Log.UserProfileErrorAsync(Logger, response, Context.RequestAborted);
+            throw new HttpRequestException("An error occurred while retrieving the user profile.");
+        }
+
+        using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync(Context.RequestAborted));
+
+        var code = payload.RootElement.GetProperty("code").GetInt32();
+
+        if (code != 0)
+        {
+            Log.UserProfileErrorCode(Logger, code, payload.RootElement.GetString("message"));
+            throw new HttpRequestException("An error occurred while retrieving the user profile.");
+        }
+
+        var principal = new ClaimsPrincipal(identity);
+        var user = payload.RootElement.GetProperty("data");
+
+        identity.AddClaim(new Claim(KookAuthenticationConstants.Claims.IsBanned, (user.GetProperty("status").GetInt32() == 10).ToString()));
```

Comment:
I'm not a fan of the `10` magic number here. Could you promote it to an inline constant with a meaningful name?

Reply:
In fact, I think it would be better to not be interpreting the API shape and creating virtual claims (like "Is banned?") here at all.

Users can implement logic like that for themselves using the OAuth events and the custom claims mapping actions.

Please remove this claim and the `IsBanned` constant.